### PR TITLE
Strings html-unescape done in fluxc

### LINKED ISSUE #6561: HTML entities displayed on different screens
I'll fix this on the FluxC side, but I'm keeping track of some screens here so we can test the PR is actually working.
screenshot-2017-08-22_13 20 42 530

### PR DESCRIPTION
**Pull request by maxme**
Fixes #6561

FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/536 must be merged before this one (and fluxc hash dependency updated then).

I removed 2 `unecapeHtml` methods in wputils (`fastUnescapeHtml` and `unescapeHTML`) that were confusing IMO.

### CODE REVIEW

**Comment by maxme** [comment]
Note: a second PR should come next with string unescaping on REST responses remaining in wpandroid.

**Comment by nbradbury** [comment]
@maxme Sorry this one has been waiting for review for so long - I'll tackle it today.

**Comment by nbradbury** [comment]
This all looks good - I'll merge it once the FluxC hash has been updated.

I should mention that the now-defunct `HtmlUtils.fastUnescapeHtml()` had an optimization where it would skip the potentially expensive `unescapeHtml4()` call if the passed string didn't contain an ampersand. This was added because I originally used this routine in an early rev of the Android reader, which often called this routine with strings that didn't contain anything to unescape.

That was most likely a premature optimization, but wanted to mention it in case we see unexplained performance issues when parsing content.

**Comment by maxme** [comment]
> That was most likely a premature optimization, but wanted to mention it in case we see unexplained performance issues when parsing content.

Honestly I didn't read the implementation of `StringEscapeUtils.unescapeHtml4` but I hope it's already optimized.

**Comment by maxme** [comment]
@nbradbury I updated FluxC dependency in a13c3d4 - ready to merge!

**Comment by nbradbury** [comment]
Looks like I spoke too soon about this being ready to merge. After using this branch I'm seeing HTML entities in the reader list that don't appear when using the `develop` branch.

screenshot_1504609913

File: libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public static String fastStripHtml(String str) {
         }
 
         // use regex to strip tags, then convert entities in the result
-        return trimStart(fastUnescapeHtml(str.replaceAll("<(.|\n)*?>", "")));
+        return trimStart(StringEscapeUtils.escapeHtml4(str.replaceAll("<(.|\n)*?>", "")));
```
Comment (maxme) [inline libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java]: 🤦‍♂️

Comment (maxme) [reply libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/HtmlUtils.java]: Fixed in 1876e69

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**